### PR TITLE
Trust management cleaning

### PR DESCRIPTION
**Pull request by pszal**
- sign all paths that are sent for a registration
- moved TRC and cert storage to a dedicated class
- TRCs stored in `.trc` files
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/466%23issuecomment-153780059%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44001871%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44001905%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44001954%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44002046%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44002152%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23issuecomment-154037212%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44125769%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44125775%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44125780%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44125828%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44125869%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23issuecomment-154382441%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44133841%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23issuecomment-155835146%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23issuecomment-153780059%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40kormat%20%22%2C%20%22created_at%22%3A%20%222015-11-04T16%3A17%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22LGTM%20in%20general.%22%2C%20%22created_at%22%3A%20%222015-11-05T11%3A35%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40kormat%20ready%20for%20another%20round%22%2C%20%22created_at%22%3A%20%222015-11-06T11%3A07%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22merging%20for%20now%2C%20the%20following%20PR%20will%20have%20unit%20tests%22%2C%20%22created_at%22%3A%20%222015-11-11T16%3A27%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20a4ff6685094e28df84883d5134034b6bbcd7945d%20lib/trust_store.py%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44001871%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Needs%20unit%20testing%22%2C%20%22created_at%22%3A%20%222015-11-05T11%3A32%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22let%20me%20postpone%20that%20until%20I%20know%20what%20exactly%20this%20class%20should%20do%20and%20contain%20%28i.e.%2C%20until%20new%20CS%20is%20ready%29%22%2C%20%22created_at%22%3A%20%222015-11-06T11%3A00%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/trust_store.py%3AL1-99%22%7D%2C%20%22Pull%20a4ff6685094e28df84883d5134034b6bbcd7945d%20lib/util.py%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44002152%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20this%20method%20still%20needed%3F%22%2C%20%22created_at%22%3A%20%222015-11-05T11%3A35%3A21Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20believe%20generator.py%20uses%20that%20%28maybe%20move%20to%20generator.py%3F%29.%20%22%2C%20%22created_at%22%3A%20%222015-11-06T11%3A02%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/util.py%3AL66-73%22%7D%2C%20%22Pull%20a4ff6685094e28df84883d5134034b6bbcd7945d%20lib/trust_store.py%2046%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44001954%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Use%20lib.util.read_file%28%29%20instead%3F%22%2C%20%22created_at%22%3A%20%222015-11-05T11%3A33%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%22%2C%20%22created_at%22%3A%20%222015-11-06T11%3A01%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/trust_store.py%3AL1-99%22%7D%2C%20%22Pull%20a4ff6685094e28df84883d5134034b6bbcd7945d%20lib/trust_store.py%2045%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44001905%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Remove%20comment%3F%22%2C%20%22created_at%22%3A%20%222015-11-05T11%3A32%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%22%2C%20%22created_at%22%3A%20%222015-11-06T11%3A00%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/trust_store.py%3AL1-99%22%7D%2C%20%22Pull%20a4ff6685094e28df84883d5134034b6bbcd7945d%20lib/trust_store.py%2055%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/466%23discussion_r44002046%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20data%20is%20only%20in%20the%20filename%2C%20and%20not%20in%20the%20contents%3F%22%2C%20%22created_at%22%3A%20%222015-11-05T11%3A33%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22for%20now%20yes%2C%20I%27ll%20add%20missing%20API%20to%20CertChain%20in%20the%20following%20PRs%22%2C%20%22created_at%22%3A%20%222015-11-06T11%3A01%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-11-06T13%3A04%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/trust_store.py%3AL1-99%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/466#issuecomment-153780059'>General Comment</a></b>
- <a href='https://github.com/pszalach'></a> @kormat
- <a href='https://github.com/kormat'></a> LGTM in general.
- <a href='https://github.com/pszalach'></a> @kormat ready for another round
- <a href='https://github.com/pszalach'></a> merging for now, the following PR will have unit tests
- [ ] <a href='#crh-comment-Pull a4ff6685094e28df84883d5134034b6bbcd7945d lib/trust_store.py 1'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/466#discussion_r44001871'>File: lib/trust_store.py:L1-99</a></b>
- <a href='https://github.com/kormat'></a> Needs unit testing
- <a href='https://github.com/pszalach'></a> let me postpone that until I know what exactly this class should do and contain (i.e., until new CS is ready)
- [ ] <a href='#crh-comment-Pull a4ff6685094e28df84883d5134034b6bbcd7945d lib/trust_store.py 45'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/466#discussion_r44001905'>File: lib/trust_store.py:L1-99</a></b>
- <a href='https://github.com/kormat'></a> Remove comment?
- <a href='https://github.com/pszalach'></a> done
- [ ] <a href='#crh-comment-Pull a4ff6685094e28df84883d5134034b6bbcd7945d lib/trust_store.py 46'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/466#discussion_r44001954'>File: lib/trust_store.py:L1-99</a></b>
- <a href='https://github.com/kormat'></a> Use lib.util.read_file() instead?
- <a href='https://github.com/pszalach'></a> done
- [ ] <a href='#crh-comment-Pull a4ff6685094e28df84883d5134034b6bbcd7945d lib/util.py 5'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/466#discussion_r44002152'>File: lib/util.py:L66-73</a></b>
- <a href='https://github.com/kormat'></a> Is this method still needed?
- <a href='https://github.com/pszalach'></a> I believe generator.py uses that (maybe move to generator.py?).
- [x] <a href='#crh-comment-Pull a4ff6685094e28df84883d5134034b6bbcd7945d lib/trust_store.py 55'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/466#discussion_r44002046'>File: lib/trust_store.py:L1-99</a></b>
- <a href='https://github.com/kormat'></a> This data is only in the filename, and not in the contents?
- <a href='https://github.com/pszalach'></a> for now yes, I'll add missing API to CertChain in the following PRs

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/466?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/466?approve=1'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/466'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: lib/trust_store.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Copyright 2015 ETH Zurich
```
Comment (kormat) [inline lib/trust_store.py:1]: Needs unit testing

File: lib/trust_store.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        self._certs = defaultdict(list)
+        self._trcs = defaultdict(list)
+        self._init_trcs()
+        self._init_certs()
+        logging.info(self._trcs)
+        logging.info(self._certs)
+
+    def _init_trcs(self):
+        for path in glob.glob("%s/*.trc" % self._dir):
+            # isd, ver = re.findall("\d+", path)[-2:]
```
Comment (kormat) [inline lib/trust_store.py]: Remove comment?

File: lib/trust_store.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            f = open(path)
```
Comment (kormat) [inline lib/trust_store.py]: Use lib.util.read_file() instead?

File: lib/trust_store.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            trc_raw = f.read()
+            f.close()
+            # trc = TRC(trc_raw)
+            self.add_trc(TRC(trc_raw))
+            logging.info("Loaded: %s" % path)
+
+    def _init_certs(self):
+        for path in glob.glob("%s/*.crt" % self._dir):
+            isd, ad, ver = map(int, re.findall("\d+", path)[-3:])
```
Comment (kormat) [inline lib/trust_store.py:51]: This data is only in the filename, and not in the contents?

File: lib/util.py [hunk 5]
```diff
@@ -66,7 +66,7 @@ def get_trc_file_path(conf_dir, isd_id, version):  # pragma: no cover
     """
     Return the TRC file path for a given ISD.
     """
-    return os.path.join(conf_dir, CERT_DIR, 'ISD%s-V%s.crt' % (isd_id, version))
+    return os.path.join(conf_dir, CERT_DIR, 'ISD%s-V%s.trc' % (isd_id, version))
```
Comment (kormat) [inline lib/util.py:69]: Is this method still needed?

File: lib/trust_store.py (hunk 1 above)
Comment (pszal) [inline lib/trust_store.py:1]: let me postpone that until I know what exactly this class should do and contain (i.e., until new CS is ready)

File: lib/trust_store.py (hunk 2 above)
Comment (pszal) [inline lib/trust_store.py]: done

File: lib/trust_store.py (hunk 3 above)
Comment (pszal) [inline lib/trust_store.py]: done

File: lib/trust_store.py (hunk 4 above)
Comment (pszal) [inline lib/trust_store.py:51]: for now yes, I'll add missing API to CertChain in the following PRs

File: lib/util.py (hunk 5 above)
Comment (pszal) [inline lib/util.py:69]: I believe generator.py uses that (maybe move to generator.py?).

File: lib/trust_store.py (hunk 4 above)
Comment (kormat) [inline lib/trust_store.py:51]: :+1:

**Pull request merged**